.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,16 @@
+name: e2e
+
+on:
+  - pull_request
```

Comment:
It can be dangerous to run on self hosted runners on each PR. And for E2E tests, maybe it's ok to just run it on main branch (and possibly release branch when we start to support LTS)

Reply:
yes, on 'pull_request' is for test this PR only, after reviewed I will change it back to main branch and tags.

Reply:
Let's change it back before merge